### DISCUSSION TITLE
What is the best practice for disabling an extension if certain environment checks are not met?

### DISCUSSION THREAD

**Discussion by NSExceptional** [microsoft/vscode-discussions#776] [Extension Development QnA] [answered]
I'm developing an extension for use in a specific project at work. I want the extension to be as absent as possible if the current workspace is not "in" this project—that is, if certain files are not present in the current workspace. This extension has commands and an activity bar item.

I noticed Copilot Chat doesn't appear in the activity bar at all somehow, unless your synced account is part of the program. Is there a way I can hide my activity bar item and disable any commands I provide like this? Then `enablement` clauses don't seem robust enough for this, unless I overlooked something

**Comment by tamuratak** [comment]
See [Activation Events](https://code.visualstudio.com/api/references/activation-events). Especially, [workspaceContains](https://code.visualstudio.com/api/references/activation-events#workspaceContains) would be what you want.

**Comment by alefragnani** [comment, marked as answer]
Hi @NSExceptional ,

As @tamuratak commented, you can use [onWorkspaceContains](https://code.visualstudio.com/api/references/activation-events#workspaceContains) activation event, and it would fit your needs, as long as you can define a `glob pattern` to the _certain files_ you need to validate in order to know that the current workspace is the kind of project you desire.

About the Activity Bar, the current contribution points (https://code.visualstudio.com/api/references/contribution-points#contributes.viewsContainers) doesn't have a way to define it as hidden, just the [Views](https://code.visualstudio.com/api/references/contribution-points#contributes.views). The Copilot Chat, on the other hand, seems to be using a different (still in proposed state) Side Bar API, as its Chat panel id is `workbench.panel.chatSidebar.copilot`. Maybe this API is responsible for the _hidden_ feature you described, but I'm not sure, because I didn't find any docs about it yet.

But, even if you are not able to hide the Activity Bar, maybe you could use the [View Welcome API](https://code.visualstudio.com/api/references/contribution-points#contributes.viewsWelcome), providing some kind of documentation / help about your extension and why it is not available in the workspace. Even, if it makes sense, give tips/shortcuts on how the user could  _enable_ the workspace to be the project you expect.

Hope this helps

**Comment by NSExceptional** [comment]
Thanks @alefragnani + @tamuratak!
